### PR TITLE
Fog-Report - Helm chart improvements for consolidating fog instances.

### PR DESCRIPTION
**Pull request by jgreat**
- Break out fog-report chart from fog-services monolith.
- Update GHA and integration tests to support multiple fog-report instances.

### Motivation

We need to be able to build fog instances that have multiple copies of fog-report, fog-view-router and fog-ledger-router with appropriate ingress and secret alignment.

### CODE REVIEW

File: .github/workflows/mobilecoin-dev-cd.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
 # Generate environment information
 ############################################
   generate-metadata:
-    if: "! startsWith(github.head_ref, 'dependabot/')"
```
Comment (jgreat) [inline .github/workflows/mobilecoin-dev-cd.yaml:38]: Cleaning up syntax so actionlint doesn't complain.

File: .github/workflows/mobilecoin-dev-cd.yaml [hunk 2]
```diff
@@ -367,13 +367,14 @@ jobs:
         - mc-core-dev-env-setup
         - mobilecoind
         - watcher
+        - fog-report
```
Comment (jgreat) [inline .github/workflows/mobilecoin-dev-cd.yaml:370]: Add fog-report to the list

File: .github/workflows/mobilecoin-workflow-dev-deploy.yaml [hunk 3]
```diff
@@ -43,10 +43,17 @@ on:
       DEV_RANCHER_TOKEN:
         description: "Rancher access token"
         required: true
+      DEV_FOG_REPORT_B_SIGNING_CERT:
```
Comment (jgreat) [inline .github/workflows/mobilecoin-workflow-dev-deploy.yaml:46]: "a" cert values are populated by the mobilecoin-workflow-dev-setup-environment.yaml action.  I'll make a second pass and see if I can make this more consistent.  For now I'm looking to minimize changes.

File: .internal-ci/helm/fog-services-config/templates/fog-report-configmap.yaml
Comment (joekottke) [inline .internal-ci/helm/fog-services-config/templates/fog-report-configmap.yaml:1]: Why is this file deleted?

File: .internal-ci/helm/fog-services-config/values.yaml [hunk 4]
```diff
@@ -32,12 +32,6 @@ fogReport:
   signingCert:
     key: ''
     crt: ''
-  configMap:
-    # https://docs.diesel.rs/diesel/r2d2/struct.Builder.html
-    POSTGRES_IDLE_TIMEOUT: '60'
-    POSTGRES_MAX_LIFETIME: '120'
-    POSTGRES_CONNECTION_TIMEOUT: '5'
-    POSTGRES_MAX_CONNECTIONS: '3'
```
Comment (joekottke) [inline .internal-ci/helm/fog-services-config/values.yaml:40]: Are these deletions pertinent?  Or are they just unused?

**Review by joekottke — APPROVED** [review]
Just curious about the deletions, and a few comments.  Non-blocking.

File: .internal-ci/helm/fog-services-config/values.yaml (hunk 4 above)
Comment (jgreat) [reply .internal-ci/helm/fog-services-config/values.yaml:40]: This moved in to the fog-report chart.  Removed this value and configmap so we don't have conflicts when applying the new fog-report chart.

File: .internal-ci/helm/fog-services-config/templates/fog-report-configmap.yaml
Comment (jgreat) [reply .internal-ci/helm/fog-services-config/templates/fog-report-configmap.yaml:1]: This moved in to the fog-report chart.  Removed this value and configmap so we don't have conflicts when applying the new fog-report chart.
